chessboard/script.js

```diff
@@ -0,0 +1,97 @@
+const board = document.querySelector(".board");
+const chess = document.querySelector(".chess");
+const bottom = document.createElement("div");
+const left = document.createElement("div");
+bottom.className = "bottom";
+left.className = "left";
+for (let i = 0; i < 8; i++) {
+  console.log(i, String.fromCharCode(i + 97));
+  const div = document.createElement("div");
+  const div1 = document.createElement("div");
+  const p = document.createElement("p");
+  const p1 = document.createElement("p");
+  p.textContent = String.fromCharCode(i + 97);
+  div.appendChild(p);
+  bottom.appendChild(div);
+  p1.textContent = 8 - i;
+  div1.appendChild(p1);
+  left.appendChild(div1);
+}
+chess.appendChild(bottom);
+chess.appendChild(left);
+console.log(board);
+for (let i = 0; i < 8; i++) {
+  const block = document.createElement("div");
+  block.className = "block";
+  for (let j = 0; j < 8; j++) {
+    const box = document.createElement("div");
+    box.className = "box";
+    box.id = `box` + (i + 1) + "" + (j + 1);
+    block.appendChild(box);
+  }
+  board.appendChild(block);
+}
+const check = (upper) => {
+  const num = upper.toString();
+  if (
+    Number(num[0]) >= 1 &&
+    Number(num[0]) <= 8 &&
+    Number(num[1]) >= 1 &&
+    Number(num[1]) <= 8
+  ) {
+    return true;
+  } else {
+    return false;
+  }
+};
+document.querySelectorAll(".box").forEach((el) => {
+  const selectedBox = el.id.slice(3, el.id.length);
```

Comment:
The code is almost similar in both the event listeners and this can be improved by creating helper functions
```

// Helper function to add or remove the 'other' class to a square
const updateSquare = (id, add) => {
  const square = document.getElementById(`box${id}`);
  if (square && isValidMove(id)) {
    square.classList[add ? 'add' : 'remove']('other');
  }
};

// Main function to highlight squares that the bishop can attack
const highlightSquares = (selectedBox, isEntering) => {
  let modifier = isEntering ? 1 : -1; // Determines whether to add or remove the highlight
  // Diagonal directions: up-left, up-right, down-left, down-right
  let directions = [-11, -9, 9, 11];  //best to have an array of valid directions

  directions.forEach((direction) => {
    let nextSquare = parseInt(selectedBox, 10);
    // Use the isValidMove function to verify if the square is within bounds before highlighting
    while (isValidMove(nextSquare)) {
      updateSquare(nextSquare, isEntering);
      nextSquare += direction * modifier;
    }
  });
};

// Attach event listeners to each square on the board, we just have to pass a boolean for the main function to manipulate the CSS classes

document.querySelectorAll(".box").forEach((el) => {
  el.addEventListener("mouseenter", () => {
    const selectedBox = el.id.slice(3); 
  console.log(el.id)
    el.classList.add("hover");
    highlightSquares(selectedBox, true); // Highlight squares when mouse enters
  });

  el.addEventListener("mouseleave", () => {
    const selectedBox = el.id.slice(3); //the ed.id returns box12, box23 etc, since we need the numbers 12, 23, only hence we are slicing the string to get the box number from the id of the box
    el.classList.remove("hover");
    highlightSquares(selectedBox, false); // Remove highlight when mouse leaves
  });
});
```